lib/finch/http1/pool.ex

```diff
@@ -175,30 +175,34 @@ defmodule Finch.HTTP1.Pool do
     # Register our pool with our module name as the key. This allows the caller
     # to determine the correct pool module to use to make the request
     {:ok, _} = Registry.register(registry, shp, __MODULE__)
-    {:ok, {registry, shp, pool_idx, metric_ref, opts}}
+
+    {:ok, {registry, shp, pool_idx, metric_ref, opts, init_activity_info()}}
   end
 
   @impl NimblePool
-  def init_worker({_name, {scheme, host, port}, _pool_idx, _metric_ref, opts} = pool_state) do
+  def init_worker(
+        {_name, {scheme, host, port}, _pool_idx, _metric_ref, opts, _actt_info} =
```

Comment:
at this point, i'm thinking its probably worth introducing a PoolState struct, the tuple is getting a bit complex. What do you think?

Reply:
I was thinking the same thing, but it will be a bigger change. 

I can do it on another PR or on this one. Let me know what you think and I can tackle it this weekend.

Reply:
Ok yeah, another PR makes sense, thanks!

Reply:
PR is open! https://github.com/sneako/finch/pull/308 as soon as it merges I will rebase this pr and fix the conflicts.